lib/app.js

```diff
@@ -49,6 +49,10 @@ async function initializeApp(app) {
     }
     provider(app.registry);
   }
+
+  app.registry.singleton.setFactory('commands[dump-object-graph]', null, () => {
+    return require('./commands/dump-object-graph');
```

Comment:
question: Why not assign an object at the top of the file?

```suggestion
    return dumpObjectGraph;
```

Reply:
I think in this case that's fine. Usually we want to do as little work as possible when registering dependency providers. The problem with requiring all the code up-front is that it slows down the server start-up - even if none of the dependencies end up being instantiated. So it's usually a good habit to lazily require the actual implementation once the factory is called (and we know for sure that it will be used).